README.md

```diff
@@ -1,88 +1,132 @@
+```http
+POST http://localhost:8081/provision HTTP/1.1
+content-type: application/json
 
+    {
+        "deviceHostName": "pi@192.168.0.6",
+        "permissions": ["temperature", "humidity"],
+        "brokerIp": "192.168.0.12",
+        "brokerUsername": "admin:e72928",
+        "brokerPassword": "admin"
+    }
 ```
-./bin/janus-cli_linux_amd64 deploy remote --agent-port <port-number> --agent-name <agent-name> -H <device-user>@<device-ip>
-```
-
-For example:
 
+### Deploy multiple hosts by config file
+
+```yaml
+
+default: # Default values will be set on agents blank fields.
+  sensors:
+  - temperature
+  - humidity
+  broker:
+    ip: "127.0.0.1"
+    username: "admin"
+    password: "admin"
+agents:
+  - hostname: pi@192.168.0.1
+    sensors:
+      - humidity
+    broker:
+      id: "800a9f"
+  - hostname: pi@192.168.0.2
+    broker:
+      id: "e72928"
 ```
-./bin/janus-cli_linux_amd64 deploy remote --agent-port 8001 --agent-name demo -H pi@192.168.0.2
-```
-
-The Aries agent will be available at port 8001 and the admin page at port 8002, such as http://192.168.0.2:8002
-
-## How to deploy an agent locally via the Ubuntu terminal
-
-To deploy locally you can run, but it will only be able to communicate with other local agents.
-
-```
-./bin/janus-cli_linux_amd64 deploy local --agent-port <port-number> --agent-name <agent-name>
-```
-
-For example:
 
+```cmd
+janus-cli deploy holder -F ./agents.yaml -p
 ```
-./bin/janus-cli_linux_amd64 deploy local --agent-port 8001 --agent-name demo
-```
+> -p refers to auto-provisioning if the janus-controller is already running 
 
-The Aries agent will be available at port 8001 and the admin page at port 8002, such as http://localhost:8002
+### Need more help using it? 
+For more details about the CLI use -h flag to get some help:
 
-If you want to have communication between local and remote devices you need to pass the network IP for the local device:
-
-```
-./bin/janus-cli_linux_amd64 deploy local --agent-port <port-number> --agent-name <agent-name> --agent-ip <agent-ip>
+```cmd
+janus-cli -h
+janus-cli deploy -h
 ```
+## Development
 
-## How to use the CLI in other operating systems
-
-To use in Windows check [here](./docs/windows.md) the doc.
+Janus was developed usign go 1.20, Docker 20.10.24 and relies over ACA-py agents and docker automation. For diagrams and more details about the implementation, check [here](./docs/implementation.md).
 
-To use in macOS check [here](./docs/apple.md) the doc.
+Before starting working it's required to run 	``` go generate ./... ``` so the docker files will be copied to the corresponding directories, required for the docker automation.
 
-Read more about the proposed features [here](./docs/readme.md)
+For building the application use Make with the command ```make build-cli```. This will generate binaries for linux, mac and windows on /bin folder. 
```

Comment:
```suggestion
For building the application use Make with the command ```make build-cli```. This will generate binaries for Linux, Mac and Windows on /bin folder. 
```